### ISSUE TITLE
type_repetition_in_bounds not always helps

### ISSUE THREAD

**Issue by tspiteri** [rust-lang/rust-clippy#4380]
For a trait like this
```rust
trait LotsOfBounds
where
    Self: Clone + Copy + Default + Ord,
    Self: Add<Output = Self> + AddAssign + Sub<Output = Self> + SubAssign,
    Self: Mul<Output = Self> + MulAssign + Div<Output = Self> + DivAssign,
{
}
```
`type_repetition_in_bounds` recommends combining all the traits, which would lead to either a very long line or to many many lines:
```rust
trait LotsOfBounds
where
    Self: Clone
        + Copy
        + Default
        + Ord
        + Add<Output = Self>
        + AddAssign
        + Sub<Output = Self>
        + SubAssign
        + Mul<Output = Self>
        + MulAssign
        + Div<Output = Self>
        + DivAssign
{
}
```

The logical grouping in the original example makes it more readable, and it is lost with the clippy suggestion.

**Comment by flip1995** [comment]
cc @xd009642 

Yeah I can see your point. I see 3 options here:
1. Leave the lint as is and users should allow it in this cases (I don't think that this would be a good solution, though)
2. Move the lint to `pedantic`
3. Make the lint configurable and add a threshold of bounds, where the lint should trigger (Maybe up to 4 or 5 bounds?)
